Dockerfile

```diff
@@ -155,16 +135,23 @@ ENV RUSTUP_HOME=/opt/rust/rustup \
 # interact with the user or fool around with TTYs.  We also set the default
 # `--target` to musl so that our users don't need to keep overriding it
 # manually.
-RUN curl https://sh.rustup.rs -sSf | \
+RUN echo "Installing Rust (${TOOLCHAIN} toolchain)" && \
+    curl https://sh.rustup.rs -sSf | \
     env CARGO_HOME=/opt/rust/cargo \
-        sh -s -- -y --default-toolchain $TOOLCHAIN --profile minimal --no-modify-path && \
+        sh -s -- -y --default-toolchain ${TOOLCHAIN} --profile minimal --no-modify-path && \
     env CARGO_HOME=/opt/rust/cargo \
         rustup component add rustfmt && \
     env CARGO_HOME=/opt/rust/cargo \
         rustup component add clippy && \
     env CARGO_HOME=/opt/rust/cargo \
         rustup target add x86_64-unknown-linux-musl
-ADD cargo-config.toml /opt/rust/cargo/config
+        # if we also install other target's build dependencies (like tools, libs, and linkers)
+        # rustup target add x86_64-unknown-linux-musl \
+        # rustup target add aarch64-unknown-linux-musl \
+        # rustup target add armv7-unknown-linux-musleabihf \
+        # rustup target add armv7-unknown-linux-musleabi
```

Comment:
We are definitely not going to try to support multiple targets in a single image. And we won't be shipping any targets that can't link `diesel` and `sqlx`. I tried this in the past and it was maintenance headache and backwards compatibility guarantee I didn't want.

Honestly, people should use `cross` unless they need C libraries that are not supported by `cross`. I've already converted as many projects as I can, and it's great. (But it won't work for everything.)